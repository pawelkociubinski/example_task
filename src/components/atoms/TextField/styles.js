import styled from 'styled-components';

export const Container = styled.div`
  display: flex;
  margin-top: 16px;
  position: relative;
`;

export const StyledTextField = styled.input`
  background-color: transparent;
  border: none;
  border-bottom: 1px solid #ddd;
  font-size: 16px;
  height: 48px;
  left: 0;
  outline: none;
  padding: 0 15px;
  position: relative;
  width: 100%;
`;

export const CoverLayer = styled.div`
  background-color: #f7f7f7;
  bottom: 0;
  height: ${({ active }) => (active ? '100%' : '0%')};
  left: 0;
  position: absolute;
  transition: all 0.3s;
  width: 100%;
  z-index: -1;
`;
