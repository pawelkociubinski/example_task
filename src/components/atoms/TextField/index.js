/* @flow */
// Dependencies
import React from 'react';
import { noop } from 'lodash';
// Custom
import { Container, CoverLayer, StyledTextField } from './styles';

type Props = {
  active: boolean,
  onBlur: () => void,
  onChange: (event: Event) => void,
  onFocus: () => void,
  readOnly?: boolean,
  value: string
};

const TextField = (props: Props) => {
  const {
    onBlur, onChange, onFocus, readOnly, value, active,
  } = props;

  return (
    <Container>
      <StyledTextField
        onBlur={onBlur}
        onChange={onChange}
        onFocus={onFocus}
        readOnly={readOnly}
        value={value}
      />
      <CoverLayer active={active} />
    </Container>
  );
};

TextField.defaultProps = {
  theme: 'neutral',
  readOnly: false,
  onChange: noop,
};

export default TextField;
