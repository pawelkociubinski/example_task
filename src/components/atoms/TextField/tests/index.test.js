import React from 'react';
import renderer from 'react-test-renderer';
import TextField from '../index';

describe('<TextField />', () => {
  it('renders correctly', () => {
    const component = renderer.create(<TextField active={false} value="value" />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
