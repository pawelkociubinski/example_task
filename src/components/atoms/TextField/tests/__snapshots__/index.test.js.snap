// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextField /> renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-top: 16px;
  position: relative;
}

.c1 {
  background-color: transparent;
  border: none;
  border-bottom: 1px solid #ddd;
  font-size: 16px;
  height: 48px;
  left: 0;
  outline: none;
  padding: 0 15px;
  position: relative;
  width: 100%;
}

.c2 {
  background-color: #f7f7f7;
  bottom: 0;
  height: 0%;
  left: 0;
  position: absolute;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  width: 100%;
  z-index: -1;
}

<div
  className="c0"
>
  <input
    className="c1"
    onBlur={undefined}
    onChange={[Function]}
    onFocus={undefined}
    readOnly={false}
    value="value"
  />
  <div
    className="c2"
  />
</div>
`;
